test/cuda/test_tc_mapper.cc

```diff
@@ -122,19 +117,7 @@ struct TcCudaMapper2DReductionStressTest : public TcCudaMapper2DReductionTest {
 
 TEST_F(TcCudaMapper2DReductionStressTest, ThreadIdy1) {
   for (int i : {1, 2, 4, 7, 8, 11, 15, 17, 24, 32, 35, 42, 64, 128, 130}) {
-    auto res = Check(i, 1);
-    if (i > 1) {
-      std::string expected = std::string("__tc::CubReduceAlongX<") +
-          std::to_string(i) + std::string(",1,1,__tc::ReductionOp::Sum>");
-      ASSERT_NE(std::string::npos, res.second.find(expected))
-          << "In resulting code:\n"
-          << res.second << "\ncould not find: " << expected;
-    } else {
-      std::string expected = "__tc::CubReduceAlongX<";
-      ASSERT_EQ(std::string::npos, res.second.find(expected))
-          << "In resulting code:\n"
-          << res.second << "\nfound unexpected: " << expected;
-    }
```

Comment:
I like how this commit removes a lot of code, but this bit seems like
it is not directly related to the caching and looks like it would be nice to keep.
Any chance of giving callers of Check access to the ExecutorType such that
this caller can extract the `cudaSource` from the `CudaTcExecutor`?

Reply:
Yes but it is happening in the larger refactoring in branch [pr/big-refactor](https://github.com/nicolasvasilache/TensorComprehensions/blob/pr/big-refactor/test/test_tc_mapper_harness-inl.h#L54)

The code says:
```
    return std::make_pair(
        std::move(outputs), std::move(pExecutor->compiledSource));
```

It seems like wasteful extra work to locally keep this behavior?
@skimo-openhub if you feel strongly that this temporary state of affairs is not good I can put in the effort.

OTOH, let this be an opportunity to freeze the codebase for pushing the refactoring?